### DISCUSSION TITLE
The SSL connection could not be established

### DISCUSSION THREAD

**Discussion by ryanbuening** [ravendb/ravendb#19799] [Q&A]
We have seen the following exception in our production environment a couple of times:

```
Raven.Client.Exceptions.RavenException: An exception occurred while contacting https://ravendbserver-example.com
System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.
 ---> System.IO.IOException: Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host..
```

We're currently not sure if some server task is cycling something causing this, or what else to look for. Are there any RavenDB logs that would help in determining anything, or any other ideas to direct our searching?

We are running RavenDB `5.4.115` in production. Raven Studio indicates that the database has been `Up for 22 days`.

Thank you.

**Comment by ayende** [comment]
Do you have `ravendbserver-example.com`  setup in your DNS?

**Reply by ryanbuening** [reply]
Yes. We did some more digging and we think this could possibly be due to socket exhaustion. We saw a different error (`System.Net.Http.HttpRequestException: Only one usage of each socket address (protocol/network address/port) is normally permitted.`) logged around the same a scheduled job runs that contacts RavenDB for thousands of users which we noticed is opening a session for each user:

```c#
public async Task<ProfileDetailsEntity?> GetProfileDetails(string userId)
{
  using var session = store.OpenAsyncSession();
  return await session.LoadAsync<ProfileDetailsEntity>(userId);
}
```

**Reply by ayende** [reply]
That shouldn't matter, since you have a connection pool for that inside of RavenDB.

Not unless you have a separate store for each user (possible if you have separate certificates for each, but rare).

Socket exhaustion is something that we previously saw, yes. Can you check the number of connections on the instance?

**Reply by ryanbuening** [reply]
> Can you check the number of connections on the instance?

Is there a way to retroactively see the number of connections when the scheduled job was ran? The job runs early in the morning and is not actively monitored.

**Reply by ayende** [reply]
Afraid not, you need to check it at the time.
You want to run `netstat -an`
